internal/controller/connector_controller.go

```diff
@@ -0,0 +1,338 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	appsv1beta1 "github.com/hstreamdb/hstream-operator/api/v1beta1"
+)
+
+const (
+	connectorfinalizerName = "connectortemplate.hstream.io/finalizer"
+)
+
+// ConnectorReconciler reconciles a Connector object
+type ConnectorReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Connector object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.0/pkg/reconcile
+func (r *ConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log = logf.FromContext(ctx)
+
+	var connector appsv1beta1.Connector
+	if err := r.Get(ctx, req.NamespacedName, &connector); err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.Error(err, "fail to fetch Connector")
+		}
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	configMapNames := []string{}
+	for _, stream := range connector.Spec.Streams {
+		configMapNames = append(configMapNames, appsv1beta1.GenConnectorConfigMapName(connector.Spec.TemplateName, false)+"-for-"+stream)
+	}
+	var configs []map[string]interface{}
+	if connector.ObjectMeta.DeletionTimestamp.IsZero() {
+		cfgs, err := r.mergePatchesIntoConfigs(ctx, connector)
+		if err != nil {
+			log.Error(err, "fail to merge patches into config")
+
+			return ctrl.Result{}, err
+		}
+		configs = cfgs
+	}
+
+	for index, configMapName := range configMapNames {
+		var connectorConfigMap corev1.ConfigMap
+		if err := r.Get(ctx, types.NamespacedName{
+			Namespace: connector.Namespace,
+			Name:      configMapName,
+		}, &connectorConfigMap); err != nil {
+			if apierrors.IsNotFound(err) {
+				configWithService := map[string]interface{}{
+					"connector": configs[index],
+					"hstream": map[string]string{
+						"serviceUrl": "hstream://" + connector.Spec.HServerEndpoint,
+					},
+				}
+
+				configJson, _ := json.Marshal(configWithService)
+
+				err = r.Create(ctx, &corev1.ConfigMap{
```

Comment:
Just create ? what happens when updating `connector` customer resource ?

Reply:
Update is not currently handled, as this PR is Just for quick verification. I will mark a TODO comment for this.

Reply:
Didn't see TODO comment

Reply:
Haven't added it yet. 🤣